scene/main/node.cpp

```diff
@@ -1237,7 +1240,7 @@ void Node::add_child(Node *p_child, bool p_legible_unique_name, bool p_internal)
 
 	_add_child_nocheck(p_child, p_child->data.name);
```

Comment:
The problem here is that you are assuming all internal nodes will come after but, in some situations, this is not desired (such as scrollbars above children controls in ScrollContainer). This system may work if you specify where you want internal children behind or after the user nodes, so you might need two separate arrays.